### PR TITLE
docs(Foundation/Icons): Updated documentation template 

### PR DESCRIPTION
**Pull request by MarcosViniciusPC**
## Description
https://jirasoftware.catho.com.br/browse/QTM-358

## Review guide
- [ ]  Documentation (yarn storybook)
- [ ] Code review

### CODE REVIEW

**Review by rapahaeru — CHANGES_REQUESTED** [review]
rebase is needed here
image

File: stories/Icon/Icon.jsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+export const ExampleIncorrectIcons = () =>
+  exampleIncorrectIcons.map((icon, index) => (
+    // eslint-disable-next-line react/no-array-index-key
+    <IconWrapper key={`${icon.props.name}_${index}`}>{icon}</IconWrapper>
+  ));
+
+export const Teste = children => <div>{children}</div>;
+
+export default ExampleIcons;
```
Comment (rapahaeru) [inline stories/Icon/Icon.jsx:47]: How about of the file has the same name of the export?

**Review by rapahaeru — CHANGES_REQUESTED** [review]

File: stories/Icon/Icon.jsx (hunk 1 above)
Comment (MarcosViniciusPC) [reply stories/Icon/Icon.jsx:47]: Interesting, I will rename

File: stories/Icon/ExampleIcons.jsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+];
+
+const exampleIncorrectIcons = [
+  <Icon name="bike" />,
+  <Icon name="emoji_smile" skin={colors.primary['900']} />,
+];
+
+export const ExampleIcons = () =>
+  exampleIcons.map((icon, index) => (
+    // eslint-disable-next-line react/no-array-index-key
```
Comment (toncabral88) [inline stories/Icon/ExampleIcons.jsx:35]: How about to include this rule to file?

File: stories/Icon/Icon.story.mdx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+- [Usage](#usage)
+- [Catalog](#catalog)
+
+## Importing Icon
+
+```js
+import { Icon } from '@catho/quantum';
+```
+
+### <a id="usage"></a>Usage
```
Comment (toncabral88) [inline stories/Icon/Icon.story.mdx:38]: How about use heading 2 here?

File: stories/Icon/Icon.story.mdx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+<Icon name="check_box" />
+<Icon name="emoji_people" skin={colors.primary['900']} />
+<Icon name="emoji_people" skin={colors.secondary['900']} />
+<Icon name="emoji_people" skin={colors.primary['700']} />
+<Icon name="emoji_people" skin={colors.warning['900']} />
+<Icon name="emoji_people" skin={colors.error['300']} />
+<Icon name="emoji_people" skin={colors.error['300']} size="large" />
+```
+
+## Unavailable Icons
```
Comment (toncabral88) [inline stories/Icon/Icon.story.mdx:62]: How about use heading 3 here and move this item as  a subitem from Catalog?

File: stories/Icon/Icon.story.mdx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+element and no icon will render.
+
+<ExampleIncorrectIcons />
+
+```js
+<Icon name="bike" />
+<Icon name="emoji_smile" skin={colors.primary['900']} />
+```
+
+### <a id="catalog"></a>Catalog
```
Comment (toncabral88) [inline stories/Icon/Icon.story.mdx:75]: How about change this guy to a heading 2?

File: stories/Icon/Icon.story.mdx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+## Available Icons
```
Comment (toncabral88) [inline stories/Icon/Icon.story.mdx:77]: How about change this guy to a heading 3?

**Review by toncabral88 — CHANGES_REQUESTED** [review]
This doc is very nice, I just think that is necessary change heading levels, my point is `Catalog` which is a heading 3 and `Available Icons` is heading 2. 

I also found a bug on catalog when I typing something and erase value the catalog doesn't restore list with all icons. This changing isn't necessary we can add it on backlog later, but feel free to solve it here, too.

File: stories/Icon/Icon.story.mdx (hunk 3 above)
Comment (MarcosViniciusPC) [reply stories/Icon/Icon.story.mdx:38]: it got better

File: stories/Icon/Catalogue.jsx [hunk 7]
```diff
@@ -21,28 +20,30 @@ class Catalogue extends Component {
     this.setState({ filtered });
   };
 
+  resetFilteredIcons = () => {
+    this.setState({ filtered: Icons });
+  };
+
```
Comment (toncabral88) [inline stories/Icon/Catalogue.jsx:26]: Nice!

**Review by toncabral88 — APPROVED** [review]

File: stories/Icon/ExampleIcons.jsx [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+import Icon from '../../components/Icon';
+import { colors } from '../../components/shared/theme';
+
+const IconWrapper = styled.div`
+  padding: 10px;
+  display: inline-block;
+`;
+
+const exampleIcons = [
+  <Icon name="info" size="small" />,
```
Comment (rapahaeru) [inline stories/Icon/ExampleIcons.jsx:12]: The default color is loading in black.
Probably the our new doc is not using Quantum Typography.
In this component, its sets the gray color on Body.

New:
image

earlier:
image

**Review by rapahaeru — APPROVED** [review]

**Review by toncabral88 — APPROVED** [review]

File: stories/Icon/ExampleIcons.jsx (hunk 8 above)
Comment (MarcosViniciusPC) [reply stories/Icon/ExampleIcons.jsx:12]: I think it's more interesting to open a task to check this case

**Pull request merged**